### PR TITLE
Mistral daily

### PR DESCRIPTION
**Pull request by Kaporos**
This adds a random daily message generated by mistral AI. 

I don't know how your streaks system and ... works so it's up to you to test the feature whenever you want to !

### CODE REVIEW

File: providers/mistral.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import requests
+import os
+def ask_mistral(prompt, max_tokens, temperature=1):
+    print("api:", os.getenv("MISTRAL_API_KEY"))
```
Comment (emecdelam) [inline providers/mistral.py:4]: Could you remove this leftover print please

File: providers/mistral.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        'max_tokens': max_tokens,
+        'temperature': 1,
+        'messages': [{
+                'role': 'user',
+                'content': prompt
+            }
+        ],
+    }
+
+    r = requests.post("https://api.mistral.ai/v1/chat/completions", headers=headers, json=json_data)
```
Comment (emecdelam) [inline providers/mistral.py:21]: It would be nice if there was a way to handle errors ( even tho none should be risen)

File: commands/streak.py [hunk 3]
```diff
@@ -105,14 +106,26 @@ async def is_admin(self,interaction: Interaction) -> bool:
...
+I want you to be creative. Speak like you would speak to teenagers. Be teaser !
+Put some smileys and life !
+Ideas of names: dumbasses, little boys, my cute engineers, young little bitches, my lolies
+And finish the message with a joke about engineers / maths or science !    
+Hello, it's time to have fun work, have you already done everything there was to do today?
+If you hadn't any exercise, have understood the lecture?
+If you hadn't any lecture, have you advanced in your group projects?
+If you can't advance in your group project you migth consider doing summaries
+If the summaries are already done you can do the exercises in advance or seeing the next chapter, going to a lecture with basic knowledge over the topic makes the lecture morerelevant
+If you can't advance any further, yo might consider preparing for the exam by redoing some exercises and reviewing important theory concept
```
Comment (emecdelam) [inline commands/streak.py:125]: This should be removed since it was the old text

File: commands/streak.py [hunk 4]
```diff
@@ -105,14 +106,26 @@ async def is_admin(self,interaction: Interaction) -> bool:
...
+                              """, 120)
```
Comment (emecdelam) [inline commands/streak.py:126]: Wouldn't 100 token be enough?

**Review by emecdelam — CHANGES_REQUESTED** [review]

File: commands/streak.py (hunk 3 above)
Comment (Kaporos) [reply commands/streak.py:125]: I got better results injecting it into the prompt

File: commands/streak.py (hunk 4 above)
Comment (Kaporos) [reply commands/streak.py:126]: Text might be cut at 100 with a half word lmao, 120 is a safety to let him finish his sentence

File: providers/mistral.py (hunk 2 above)
Comment (Kaporos) [reply providers/mistral.py:21]: yess

**Pull request merged**